R/generate_prob_nowcasts_df.R

```diff
@@ -0,0 +1,103 @@
+#' Generate probabilistic nowcast dataframe
+#'
+#' This function ingests a list of matrices that represent expected observed
+#'    nowcasts, and generates a long tidy dataframe indexed by time and delay.
+#'
+#' @param list_of_nowcasts List of  matrices of expected
+#'    observed reporting squares.
+#' @importFrom cli cli_abort
+#' @returns `nowcast_df` Dataframe containing observations and expected
+#'    observed nowcasts indexed by reference time and delay.
+#' @export
+#'
+#' @examples
+#' point_nowcast <- matrix(
+#'   c(
+#'     80, 50, 25, 10,
+#'     100, 50, 30, 20,
+#'     90, 45, 25, 17.8,
+#'     80, 40, 23.2, 15.9,
+#'     70, 35, 20.3, 19.9
+#'   ),
+#'   nrow = 5,
+#'   byrow = TRUE
+#' )
+#'
+#' list_of_exp_obs_nowcast <- add_obs_errors_to_nowcast(
+#'   comp_rep_square = point_nowcast,
+#'   disp = c(8, 1.4, 4),
+#'   n_draws = 10
+#' )
+#'
+#' nowcast_df <- generate_prob_nowcast_df(
+#'   list_of_nowcasts = list_of_exp_obs_nowcast
+#' )
+generate_prob_nowcast_df <- function(list_of_nowcasts) {
```

Comment:
it doesn't really generate following the naming logic you have used elsewhere? what about nowcast_list_to_df or similar?

Reply:
yeah the other is more explicit (but could also maybe be named `rep_matrix_to_df()`?

Reply:
Changed to `nowcast_list_to_df()`, left the other for now